### PR TITLE
fix requirements checking code

### PR DESCRIPTION
**Pull request by spiderxm**

### CODE REVIEW

**Comment by Ali-Razmjoo** [comment]
hold on to this; I will send some updates.

**Comment by Ali-Razmjoo** [comment]
@spiderxm can you fix the PR base on this? we used this in nettacker project

* requirements.txt
```
argparse==1.4.0
netaddr==0.8.0
ipaddr==2.2.0
requests==2.25.1
click==8.0.1
paramiko==2.7.2
texttable==1.6.3
PySocks==1.7.1 # library_name=socks # module name is not equal to socks name; this is required to be checked on startup
pyOpenSSL==20.0.1 # library_name=OpenSSL
flask==2.0.1
lockfile==0.12.2
bs4==0.0.1
pyspf==2.0.14 # library_name=spf
SQLAlchemy>=1.3.0 # library_name=sqlalchemy
canari==3.3.10
py3DNS==3.2.1 # library_name=DNS
censys==2.0.3
shodan==1.25.0
pycurl==7.43.0.6
xmltodict==0.12.0
rule==0.1.1
mechanize==0.4.5
scapy==2.4.5
psycopg2==2.9.1
numpy==1.21.1
```


* dependency check function
```
    external_modules = open(default_config["requirements_path"]).read().split('\n')

    for module_name in external_modules:
        try:
            __import__(
                module_name.split('==')[0] if 'library_name=' not in module_name
                else module_name.split('library_name=')[1].split()[0]
            )
        except Exception:
            __die_failure(
                "pip3 install -r requirements.txt ---> " +
                module_name + " not installed!"
            )
```

File: config.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
     for name, num in vars(socket).items()
     if name.startswith("IPPROTO")
 }
-real_machine_ip_address = socket.gethostbyname(socket.gethostname())
+real_machine_ip_address = socket.gethostbyname('')
```
Comment (dhirensr) [inline config.py:16]: why is this changed?

Comment (spiderxm) [reply config.py]: @dhirensr Sorry accidentally committed that code. It was giving some error like no host name present. So I fixed it in local. Removed it.

**Comment by Ali-Razmjoo** [comment]
it should have been merge to development but it's ok, we can go live with this branch later.
